`yarn develop` fails [eng-web]

I'm on macOS.

```
yarn run v1.22.17
$ gatsby develop
success compile gatsby files - 0.356s
...
success onPreExtractQueries - 0.000s

 ERROR #85923  GRAPHQL

There was an error in your GraphQL query:

Cannot query field "github" on type "Query".

If you don't expect "github" to exist on the type "Query" it is most likely a typo.
However, if you expect "github" to exist there are a couple of solutions to common problems:

- If you added a new data source and/or changed something inside gatsby-node.js/gatsby-config.js, please try a restart of your development server
- The field might be accessible in another subfield, please try your query in GraphiQL and use the GraphiQL explorer to see which fields you can query and
what shape they have
- You want to optionally use your field "github" and right now it is not used anywhere. Therefore Gatsby can't infer the type and add it to the GraphQL
schema. A quick fix is to add at least one entry with that field ("dummy content")

It is recommended to explicitly type your GraphQL schema if you want to use optional fields. This way you don't have to add the mentioned "dummy content".
Visit our docs to learn how you can define the schema for "Query":
https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization#creating-type-definitions

File: src/utils/hooks/useStars.tsx:9:9

failed extract queries from components - 1.135s
success write out requires - 0.017s
success run static queries - 0.025s - 4/4 161.63/s
success run page queries - 0.022s - 3/3 137.39/s
success Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs - 0.245s - 2/2 8.17/s

 ERROR 

Module not found: Error: Can't resolve '../../../public/page-data/sq/d/1164568807.json' in '.../mlem.ai/src/utils/hooks'



  ModuleNotFoundError: Module not found: Error: Can't resolve '../../../public/page-data/sq/d/1164568807.json' in '.../mlem.ai/src/utils/hooks'
  
  - Compilation.js:2016 
    [mlem.ai]/[webpack]/lib/Compilation.js:2016:28
  
...
  
  - async.js:6858 Object.parallel
    [mlem.ai]/[neo-async]/async.js:6858:9
  
  - NormalModuleFactory.js:873 NormalModuleFactory._resolveResourceErrorHints
    [mlem.ai]/[webpack]/lib/NormalModuleFactory.js:873:12
  
  - NormalModuleFactory.js:834 
    [mlem.ai]/[webpack]/lib/NormalModuleFactory.js:834:18
  

not finished Building development bundle - 1.008s

error Command failed with exit code 1.
```

--- a/gatsby-config.js
+++ b/gatsby-config.js
@@ -100,19 +100,6 @@ const plugins = [
   'gatsby-plugin-meta-redirect'
 ]
 
-process.env.GITHUB_TOKEN &&
-  plugins.push({
-    resolve: 'gatsby-source-graphql',
-    options: {
-      typeName: 'GitHub',
-      fieldName: 'github',
-      url: 'https://api.github.com/graphql',
-      headers: {
-        Authorization: `Bearer ${process.env.GITHUB_TOKEN}`
-      }
-    }
-  })
-
 module.exports = {
   plugins,
   siteMetadata

--- a/gatsby-node.js
+++ b/gatsby-node.js
@@ -1,5 +1,6 @@
 const Prism = require('prismjs')
 const loadLanguages = require('prismjs/components/')
+const { graphql } = require('@octokit/graphql')
 
 const terminalSlides = require('./content/home-slides')
 

--- a/gatsby-node.js
+++ b/gatsby-node.js
@@ -80,7 +80,7 @@ exports.createResolvers = async ({ createResolvers }) => {
             const query = await graphql(
               `
                 {
-                  repository(owner: "iterative", name: "dvc") {
+                  repository(owner: "iterative", name: "mlem") {
                     stargazers {
                       totalCount
                     }

--- a/gatsby-node.js
+++ b/gatsby-node.js
@@ -52,3 +52,49 @@ exports.sourceNodes = ({ actions, createNodeId, createContentDigest }) => {
     createNode(node)
   })
 }
+
+exports.createSchemaCustomization = async ({
+  actions: { createTypes },
+  schema
+}) => {
+  createTypes(
+    schema.buildObjectType({
+      name: 'StaticGithubData',
+      fields: {
+        stars: 'String!'
+      }
+    })
+  )
+}
+
+exports.createResolvers = async ({ createResolvers }) => {
+  createResolvers({
+    Query: {
+      staticGithubData: {
+        type: 'StaticGithubData',
+        async resolve() {
+          console.log('Resolving GitHub Data')
+          const { GITHUB_TOKEN } = process.env
+          if (GITHUB_TOKEN) {
+            const query = await graphql(
+              `
+                {
+                  repository(owner: "iterative", name: "dvc") {
+                    stargazers {
+                      totalCount
+                    }
+                  }
+                }
+              `,
+              { headers: { authorization: `token ${GITHUB_TOKEN}` } }
+            )
+
+            const stars = query.repository.stargazers.totalCount
+            return { stars }
+          }
+          return { stars: 8888 }
+        }
+      }
+    }
+  })
+}

--- a/gatsby-node.js
+++ b/gatsby-node.js
@@ -74,7 +74,6 @@ exports.createResolvers = async ({ createResolvers }) => {
       staticGithubData: {
         type: 'StaticGithubData',
         async resolve() {
-          console.log('Resolving GitHub Data')
           const { GITHUB_TOKEN } = process.env
           if (GITHUB_TOKEN) {
             const query = await graphql(

--- a/package.json
+++ b/package.json
@@ -39,7 +39,6 @@
     "gatsby-remark-images": "^6.14.0",
     "gatsby-remark-prismjs": "^6.12.1",
     "gatsby-source-filesystem": "^4.1.0",
-    "gatsby-source-graphql": "^4.15.0",
     "gatsby-transformer-sharp": "^4.1.0",
     "node-cache": "^5.1.2",
     "npm-run-all": "^4.1.5",
@@ -49,8 +48,8 @@
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
     "react-helmet": "^6.1.0",
-    "serve-handler": "^6.1.3",
     "repo-link-check": "^0.11.0",
+    "serve-handler": "^6.1.3",
     "tailwindcss": "3.0.23",
     "typed.js": "^2.0.12"
   },

--- a/src/utils/hooks/useStars.tsx
+++ b/src/utils/hooks/useStars.tsx
@@ -3,33 +3,30 @@ import { useStaticQuery, graphql } from 'gatsby'
 import fetch from 'isomorphic-fetch'
 
 export default function useStars(): number | null {
-  const staticStars =
-    useStaticQuery(graphql`
-      query GithubStarsQuery {
-        github {
-          repository(name: "mlem", owner: "iterative") {
-            stargazerCount
-          }
-        }
+  const staticStars = useStaticQuery(graphql`
+    query GithubStarsQuery {
+      staticGithubData {
+        stars
       }
-    `).github.repository.stargazerCount || 888
+    }
+  `).staticGithubData.stars
 
   // Maintain an updatable state so we can update stars on delivery
   const [stars, setStars] = useState(staticStars)
 
-  // Run an IIFE to update from the server on the client side.
+  // Update on the client side
   useEffect(() => {
-    ;(async (): Promise<void> => {
-      const res = await fetch(`/api/github/stars`)
+    fetch(`/api/github/stars`).then(res => {
       if (res.status === 200) {
-        const json = await res.json()
-        setStars(json.stars)
+        res.json().then(json => {
+          setStars(json.stars)
+        })
       } else {
         console.warn(
           `Stars update response status was ${res.status}! Using static value.`
         )
       }
-    })()
+    })
   }, [])
 
   return stars

